### PR TITLE
Add Sync Support.

### PR DESCRIPTION
**Pull request by Natim**
This is tracking the adding of the Sync Support.

### CODE REVIEW

**Comment by Natim** [comment]
[addon-2.0.0a3.zip](https://github.com/mozilla/notes/files/1379971/addon-2.0.0a3.zip)

**Comment by Natim** [comment]
[addon-2.0.0a4.zip](https://github.com/mozilla/notes/files/1382936/addon-2.0.0a4.zip)

**Comment by Natim** [comment]
[addon-2.0.0a6.zip](https://github.com/mozilla/notes/files/1417782/addon-2.0.0a6.zip)

**Comment by Natim** [comment]
[addon-2.0.0a7.zip](https://github.com/mozilla/notes/files/1421470/addon-2.0.0a7.zip)

**Comment by Natim** [comment]
[addon-2.0.0a8.zip](https://github.com/mozilla/notes/files/1421602/addon-2.0.0a8.zip)

**Comment by Natim** [comment]
[addon-2.0.0a9.zip](https://github.com/mozilla/notes/files/1421786/addon-2.0.0a9.zip)

**Comment by vladikoff** [comment]
Updated to ` "fxa-crypto-relier": "1.0.3",` , make sure to `npm install` ⚠️

File: src/background.js [hunk 1]
```diff
@@ -36,14 +48,75 @@ function sendMetrics(event, context = {}) {
...
+      access_token: loginDetails.access_token,
+      refresh_token: loginDetails.refresh_token,
+      key,
+      metadata: {
+        server: FXA_OAUTH_SERVER,
+        client_id: FXA_CLIENT_ID,
+        scope: FXA_SCOPES
+      }
+    };
+    console.log('Login succeeded', credentials);
```
Comment (vladikoff) [inline src/background.js]: let's not forget to remove these extra logs before merge

File: package.json [hunk 2]
```diff
@@ -11,7 +11,7 @@
     "fxa-crypto-relier": "mozilla/fxa-crypto-relier#d45e814bc285ccdbc12783324407ae943915637f",
     "jose-jwe-jws": "^0.1.5",
     "kinto": "^10.0.0",
-    "kinto-http": "^4.3.4",
+    "kinto-http": "4.3.4",
```
Comment (Natim) [inline package.json]: We shouldn't do that especially since we are fixing a bunch of bugs there at the moment.

Comment (vladikoff) [reply package.json]: Ref: https://github.com/Kinto/kinto-http.js/issues/259 

Yea sorry, it was blocking people trying to build locally and try scoped keys. We should unlock it

**Pull request closed without merging**